test/auth_internal_SUITE.erl

```diff
@@ -16,17 +16,22 @@ init_per_suite(C) ->
                                                    internal => #{},
                                                    password => #{format => scram,
                                                                  scram_iterations => 10}}),
-    mongoose_domain_core:start([{domain(), host_type()}], []),
     ejabberd_auth_internal:start(host_type()),
     C.
 
 end_per_suite(_C) ->
     ejabberd_auth_internal:stop(host_type()),
-    mongoose_domain_core:stop(),
     mongoose_config:unset_opt({auth, host_type()}),
     mnesia:stop(),
     mnesia:delete_schema([node()]).
 
+init_per_testcase(_TC, Config) ->
+    mongoose_domain_core:start_link([{domain(), host_type()}], []),
```

Comment:
Please add a comment that `mongoose_domain_core` "stops" because every TC runs in a separate process and linking ensure termination of `mongoose_domain_core`, and parallel execution of TCs is not possible.